cmd/src/snapshot_databases.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/output"
+	"gopkg.in/yaml.v3"
+
+	"github.com/sourcegraph/src-cli/internal/pgdump"
+)
+
+func init() {
+	usage := `'src snapshot databases' generates commands to export Sourcegraph database dumps.
+Note that these commands are intended for use as reference - you may need to adjust the commands for your deployment.
+
+USAGE
+	src [-v] snapshot databases <pg_dump|docker|kubectl> [--targets=<docker|k8s|"targets.yaml">]
+
+TARGETS FILES
+	Predefined targets are available based on default Sourcegraph configurations ('docker', 'k8s').
+	Custom targets configuration can be provided in YAML format with '--targets=target.yaml', e.g.
+
+		primary:
+			entity: ...
+			dbname: ...
+			username: ...
+			password: ...
+		codeintel:
+			# same as above
+		codeinsights:
+			# same as above
+`
+	flagSet := flag.NewFlagSet("databases", flag.ExitOnError)
+	targetsKeyFlag := flagSet.String("targets", "auto", "predefined targets ('docker' or 'k8s'), or a custom targets.yaml file")
+
+	snapshotCommands = append(snapshotCommands, &command{
+		flagSet: flagSet,
+		handler: func(args []string) error {
+			if err := flagSet.Parse(args); err != nil {
+				return err
+			}
+			out := output.NewOutput(flagSet.Output(), output.OutputOpts{Verbose: *verbose})
+
+			var builder string
+			if len(args) > 0 {
+				builder = args[0]
+			}
+
+			targetKey := "docker"
+			var commandBuilder pgdump.CommandBuilder
+			switch builder {
+			case "pg_dump", "":
+				commandBuilder = func(t pgdump.Target) (string, error) {
+					return pgdump.Command(t), nil
+				}
+			case "docker":
+				commandBuilder = func(t pgdump.Target) (string, error) {
+					// docker exec -it pgsql sh -c 'pg_dump --no-owner --format=p --no-acl --username=sg --dbname=sg'
+					return fmt.Sprintf("docker exec -it %s sh -c '%s'", t.Target, pgdump.Command(t)), nil
+				}
+			case "kubectl":
+				targetKey = "k8s"
+				commandBuilder = func(t pgdump.Target) (string, error) {
+					// docker exec -it pgsql sh -c 'pg_dump --no-owner --format=p --no-acl --username=sg --dbname=sg'
```

Comment:
```suggestion
					// kubect exec -it pgsql -- bash -c 'pg_dump --no-owner --format=p --no-acl --username=sg --dbname=sg'
```

Reply:
oops, did not intend to keep this docstring